packages/devextreme/js/__internal/ui/speech_to_text/speech_to_text.ts

```diff
@@ -0,0 +1,261 @@
+
+export const SPEECH_TO_TEXT_CLASS = 'dx-speech-to-text';
+export const SPEECH_TO_TEXT_LISTENING_CLASS = 'dx-speech-to-text-listening';
+
+const DEFAULT_INITIAL_ICON = 'micoutline';
+const DEFAULT_LISTENING_ICON = 'stopfilled';
+
+const ACTIONS: (keyof SpeechToTextActions)[] = [
+  'onStartClick',
+  'onStopClick',
+  'onResult',
+  'onError',
+];
+
+class SpeechToText extends Widget<SpeechToTextProperties> {
+  private _button!: Button;
+
+  private _state!: SpeechToTextState;
+
+  private _actions!: SpeechToTextActions;
+
+  _getDefaultOptions(): SpeechToTextProperties {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return extend(super._getDefaultOptions(), {
+      stylingMode: 'contained',
+      type: 'default',
+      startIcon: DEFAULT_INITIAL_ICON,
+      stopIcon: DEFAULT_LISTENING_ICON,
+      startText: '',
+      stopText: '',
+      customSpeechRecognizer: {
+        enabled: false,
+        isListening: false,
+      },
+      onStartClick: undefined,
+      onStopClick: undefined,
+      onResult: undefined,
+      onError: undefined,
+    });
+  }
+
+  _init(): void {
+    this._actions = {};
+    super._init();
+    this._handleCustomEngineState();
+    this._createActions();
+  }
+
+  _initMarkup(): void {
+    super._initMarkup();
+    this.$element().addClass(SPEECH_TO_TEXT_CLASS);
+    this._renderButton();
+    this._updateButtonState();
+  }
+
+  private _createActions(): void {
+    ACTIONS.forEach((action) => {
+      this._setAction(action);
+    });
+  }
+
+  private _setAction(action: keyof SpeechToTextActions): void {
+    this._actions[action] = this._createActionByOption(action, {
+      excludeValidators: ['disabled', 'readOnly'],
+    }) || noop;
+  }
+
+  private _renderButton(): void {
+    this._button = this._createComponent<Button, ButtonProperties>(
+      this.$element(),
+      Button,
+      this._getButtonOptions(),
+    );
+  }
+
+  private _getButtonOptions(): ButtonProperties {
+    const {
+      activeStateEnabled,
+      focusStateEnabled,
+      hoverStateEnabled,
+      stylingMode,
+      type,
+      disabled,
+      width,
+      height,
+    } = this.option();
+
+    return {
+      stylingMode,
+      type,
+      disabled,
+      width,
+      height,
+      activeStateEnabled,
+      focusStateEnabled,
+      hoverStateEnabled,
+      icon: this._getCurrentIcon(),
+      text: this._getCurrentText(),
+      onClick: (e: ClickEvent): void => {
+        this._buttonClickHandler(e);
+      },
+    };
+  }
+
+  private _getCurrentIcon(): string | undefined {
+    const { startIcon, stopIcon } = this.option();
+
+    return this._isListening() ? stopIcon : startIcon;
+  }
+
+  private _getCurrentText(): string {
+    const { startText, stopText } = this.option();
+
+    return this._isListening()
+      ? stopText ?? ''
+      : startText ?? '';
+  }
+
+  private _buttonClickHandler(e: ClickEvent): void {
```

Comment:
let's state with a verb as in other places
```suggestion
  private _handleButtonClick(e: ClickEvent): void {
```

Reply:
[9e2a7752de838dac87a3a6aa040186168a64c19c]